neuro_flow/expr.py

```diff
@@ -148,8 +156,44 @@ async def from_json(ctx: CallCtx, arg: str) -> TypeT:
     return cast(TypeT, json.loads(arg))
 
 
+def _check_has_needs(ctx: CallCtx, func_name: str) -> None:
+    try:
+        ctx.root.lookup("needs")
+    except LookupError:
+        raise ValueError(f"{func_name}() is only available inside task definition")
+
+
+async def always(ctx: CallCtx) -> bool:
+    _check_has_needs(ctx, func_name="always")
+    return True
+
+
+async def success(ctx: CallCtx) -> bool:
+    _check_has_needs(ctx, func_name="success")
+    needs = ctx.root.lookup("needs")
+    assert isinstance(needs, MappingT)
+    for dependency in needs:
+        dep_ctx = needs[dependency]
+        if dep_ctx.result != JobStatus.SUCCEEDED:  # type: ignore
+            return False
+    return True
+
+
+async def failure(ctx: CallCtx) -> bool:
+    _check_has_needs(ctx, func_name="failure")
+    return not success(ctx)
+
+
 FUNCTIONS = _build_signatures(
-    len=alen, nothing=nothing, fmt=fmt, keys=akeys, to_json=to_json, from_json=from_json
+    len=alen,
+    nothing=nothing,
+    fmt=fmt,
+    keys=akeys,
+    to_json=to_json,
+    from_json=from_json,
+    always=always,
+    success=success,
+    failure=failure,
```

Comment:
Is there reason for not implementing `cancelled()` function?

Reply:
I didn't found any way to cancel started workflow, so at first I think that we do not have concept of cancelled task. But now we do (task with `enable: False`), so thanks for suggestion, I'll add it.

Reply:
I have added `cancelled()`, but I am not sure about it:
I think that it should evaluate to `True` only when the workflow is canceled. I mean, after running something like `neuro-flow cancel batch-id`. I do not think that `task-b` should always in the following workflow:
```
kind: batch
tasks:
  - id: task_a
    enable: false
    image: ubuntu
    preset: cpu-small
    bash: echo "Task A"

  - id: task-b
    needs: [task-a]
    enable: ${{ cancelled() }}
    image: ubuntu
    preset: cpu-small
    bash: echo "Task B"
```
Therefore, we should probably avoid adding `cancelled()` function until the canceling of workflows will be available.

Reply:
Found this issue https://github.com/neuromation/neuro-flow/issues/68. I think that `cancelled()` should raise `NotImplementedError` until that issue is closed.